runtime/compiler/env/DependencyTable.cpp

```diff
@@ -60,17 +178,42 @@ TR_AOTDependencyTable::classLoadEvent(TR_OpaqueClassBlock *clazz, bool isClassLo
    catch (std::exception&)
       {
       deactivateTable();
+      return;
+      }
+
+   resolvePendingLoads();
+   }
+
+void
+TR_AOTDependencyTable::checkForSatisfaction(PersistentUnorderedSet<MethodEntryRef> waitingMethods)
+   {
+   for (auto &entry: waitingMethods)
+      {
+      if (entry->second._remainingDependencies)
+         _pendingLoads.insert(entry);
+      else
+         --entry->second._remainingDependencies;
       }
    }
 
 void
 TR_AOTDependencyTable::classLoadEventAtOffset(J9Class *ramClass, uintptr_t offset, bool isClassLoad, bool isClassInitialization)
    {
-   auto entry = getOffsetEntry(offset, isClassLoad);
-   TR_ASSERT(entry || !isClassLoad, "Class %p offset %lu initialized without loading");
+   auto offsetEntry = getOffsetEntry(offset, isClassLoad);
+   TR_ASSERT(offsetEntry || !isClassLoad, "Class %p offset %lu initialized without loading", ramClass, offset);
```

Comment:
This isn't accurate - I check that `ramClass` matches its cached version for loads only, so if this isn't a load, I need to check that `ramClass` is in `offsetEntry->_loadedClasses`. If it isn't, then it was intentionally not added when it was loaded.